code/nuxeo-apidoc-jsf/src/main/java/org/nuxeo/apidoc/seam/introspection/SeamComponentInfoImpl.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * (C) Copyright 2006-2010 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thierry Delprat
+ */
+package org.nuxeo.apidoc.seam.introspection;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.nuxeo.apidoc.api.BaseNuxeoArtifact;
+import org.nuxeo.apidoc.seam.api.SeamComponentInfo;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreType;
+
+@JsonIgnoreType
+public class SeamComponentInfoImpl extends BaseNuxeoArtifact implements SeamComponentInfo {
+
+    protected String name;
+
+    protected String scope;
+
+    protected String precedence;
+
+    protected String className;
+
+    protected final List<String> interfaceNames = new ArrayList<>();
+
+    protected String version;
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getScope() {
+        return scope;
+    }
+
+    @Override
+    public String getPrecedence() {
+        return precedence;
+    }
+
+    @Override
+    public String getClassName() {
+        return className;
+    }
+
+    public void addInterfaceName(String name) {
+        if (!interfaceNames.contains(name)) {
```

Comment:
Use a `Set` instead of a List in that case.

Reply:
agreed, but this is legacy code, not worth updating on deprecated JSF imho